pathd/path_pcep_controller.c

```diff
@@ -397,15 +442,15 @@ int pcep_thread_send_report_callback(struct thread *t)
 
 /* ------------ Controller Thread Timer Handler ------------ */
 
-int schedule_thread(struct ctrl_state *ctrl_state, int pcc_id,
-		    enum pcep_ctrl_timer_type type, uint32_t delay,
-		    void *payload, struct thread **thread)
+int schedule_thread_timer(struct ctrl_state *ctrl_state, int pcc_id,
+			  enum pcep_ctrl_timer_type type, uint32_t delay,
+			  void *payload, struct thread **thread)
 {
 	assert(thread != NULL);
 
 	struct pcep_ctrl_timer_data *data;
 
-	data = XCALLOC(MTYPE_PCEP, sizeof(*data));
+	data = XCALLOC(MTYPE_PCEP, sizeof(struct pcep_ctrl_timer_data));
```

Comment:
Why this change ? I feel that duplicating the type is not an advantage here.

Reply:
I had some memory problems, and was trying different things. I'll put it back.